src/index.js

```diff
@@ -78,22 +78,30 @@ router.get('/', ctx => {
 app.use(koaStatic(path.join(__dirname, '../static/')));
 app.use(router.routes(), router.allowedMethods());
 
-const apolloServer = new ApolloServer({
+export const apolloServer = new ApolloServer({
   schema,
   introspection: true, // Allow introspection in production as well
   playground: true,
-  context: ({ ctx }) => ({
-    loaders: new DataLoaders(), // new loaders per request
-    user: ctx.state.user,
-
-    // userId-appId pair
-    //
-    userId:
-      ctx.appId === 'WEBSITE' || ctx.appId === 'DEVELOPMENT_FRONTEND'
-        ? (ctx.state.user || {}).id
-        : ctx.query.userId,
-    appId: ctx.appId,
-  }),
+  context: async ({ ctx }) => {
+    let currentUser = ctx.state.user || {};
+    if (ctx.appId) {
+      const { user } = await createOrUpdateUser({
+        appUserId: ctx.query.userId,
```

Comment:
Does this work for `ctx.appId` is `WEBSITE` (browser app users) ?

Reply:
oh, there's no `ctx.query.userId` for web users?

Reply:
Nope. For web app users it relies on `ctx.state.user`, which is populated by `passport.deserializeUser` callback, which passport.js will call when de-serialize data (user id) in cookie to `ctx.state.user`.

Ref: https://g0v.hackmd.io/ZcoUOX_-RQSkJyl5xz4_Zg#Identifying-the-app-and-user